src/NeoSharp.Core/SmartContract/Invocation/InvocationProcess.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using NeoSharp.Core.Blockchain;
+using NeoSharp.Core.Models;
+using NeoSharp.Core.SmartContract.ContractParameters;
+using NeoSharp.Core.Types;
+using NeoSharp.VM;
+
+namespace NeoSharp.Core.SmartContract
+{
+    public class InvocationProcess
+    {
+        private IExecutionEngine _executionEngine;
+
+        public InvocationProcess(IExecutionEngine executionEngine)
+        {
+            _executionEngine = executionEngine;
+        }
+
+        /// <summary>
+        /// Invokes the specified Contract using the provided parameters.
+        /// </summary>
+        /// <returns>The invoke.</returns>
+        /// <param name="scriptHash">Script hash.</param>
+        /// <param name="contractParameters">Contract parameters.</param>
+        public InvocationResult Invoke(UInt160 scriptHash, ContractParameter[] contractParameters)
+        {
+            var invocationScript = BuildInvocationScript(scriptHash, contractParameters);
+            var invocationResult = RunInvocationOnVM(invocationScript);
+            return invocationResult;
+        }
+
+        /// <summary>
+        /// Concatenates the scripts and runs it in a single invocation.
+        /// </summary>
+        /// <returns>The multiple.</returns>
+        /// <param name="contractParameterMap">Contract parameter map.</param>
+        public InvocationResult InvokeMultiple(Dictionary<UInt160, ContractParameter[]> contractParameterMap)
+        {
+            List<byte> invocationScript = new List<byte>();
+
+            foreach(var scriptHash in contractParameterMap.Keys){
+                ContractParameter[] parameters = null;
+                if(contractParameterMap.TryGetValue(scriptHash, out parameters)){
+                    byte[] script = BuildInvocationScript(scriptHash, parameters);
+                    invocationScript.AddRange(script);    
+                }else{
+                    throw new ArgumentException(nameof(contractParameterMap));
+                }
+            }
+
+            var invocationResult = RunInvocationOnVM(invocationScript.ToArray());
+            return invocationResult;
+        }
+
+        /// <summary>
+        /// Builds the invocation script.
+        /// </summary>
+        /// <returns>The invocation script.</returns>
+        /// <param name="scriptHash">Script hash.</param>
+        /// <param name="parameters">Parameters.</param>
+        private byte[] BuildInvocationScript(UInt160 scriptHash, ContractParameter[] parameters)
+        {
+            byte[] executionScript;
+
+            using (var sb = new ScriptBuilder())
+            {
+                for (int i = parameters.Length - 1; i >= 0; i--)
+                    parameters[i].PushIntoScriptBuilder(sb);
```

Comment:
`sb.Push(ContactParameter parameter) `

Reply:
But you said you didn't want this method to be an ScriptBuilder extension 🤔